homeassistant/components/roborock/scene.py

```diff
@@ -0,0 +1,75 @@
+"""Support for Roborock scene."""
+
+from __future__ import annotations
+
+import asyncio
+from typing import Any
+
+from roborock.containers import HomeDataScene
+
+from homeassistant.components.scene import Scene as SceneEntity
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity import EntityDescription
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import RoborockConfigEntry
+from .coordinator import RoborockDataUpdateCoordinator
+from .entity import RoborockEntity
+
+
+async def _setup_coordinator(
+    coordinator: RoborockDataUpdateCoordinator,
+) -> list[SceneEntity]:
+    return [
+        RoborockSceneEntity(coordinator, scene)
```

Comment:
I think this function should just return the scene data, and then create the entity in `async_setup_entry`. Name the fucntion `get_scenes` and have it return scenes. 

(The name `coordinator_entities` doesn't really mean anything, and this function `_setup_coordinator` has an ambiguous name given the coordinators are already setup)

Reply:
I renamed the functions, that was an oversight on my end. Regarding the creation in the async setup entry - because the creations are coupled to both scene data and the coordinator, I'd rather keep it like today, but if you prefer to do it your way, I will need to make the get scene function return a tuple of coordinator and scene data, which I think is less clean. Let me know what you think